.github/workflows/pr_title.yml

```diff
@@ -0,0 +1,109 @@
+name: PR Title Check
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize, reopened]
+jobs:
+  commitlint:
+    name: PR title / description conforms to semantic-release
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "14"
+      - run: npx commitlint --verbose <<< $COMMIT_MSG
+        env:
+          COMMIT_MSG: >
+            ${{ github.event.pull_request.title }}
+            
+            ${{ github.event.pull_request.body }}
+      - if: failure()
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const message = `Substrait follows the [Conventional Commits specification]\
+              (https://www.conventionalcommits.org/en/v1.0.0/) for release automation.
+              The PR title and description are used as the merge commit message.
+              
+              Please update your PR title and description to match the specification.`
+            // Get list of current comments
+            const list_comments = github.rest.issues.listComments({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.issue.number
+            })
+            const comments = await github.paginate(list_comments)
+            // Check if this job already commented
+            for (const comment of comments) {
+              if (comment.body === message) {
+                return // Already commented
+              }
+            }
+            // Post the comment about Conventional Commits
+            github.rest.issues.createComment({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.issue.number,
+              body: message
+            })
+            core.setFailed('Invalid PR title')
+  breaking:
+    name: Ensure breaking changes are labeled in description
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
Because this job requires a checkout we should probably move this back to the other PR workflow (with a normal `pull_request` trigger) and skip the PR comment (so it doesn't require access to the secrets).
https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

Reply:
I moved it to a separate workflow (using the `pull_request` target) because I want it to be triggered on a PR edit.